src/read_utils.jl

```diff
@@ -1,3 +1,124 @@
+    kvecs = KVec.(eachrow(@view dispersion_data[:,2:2+(D-1)])) ##No longer works? for new version of Crystalline
+    kidxs = eachindex(kvecs)
+    Nk = length(kidxs)
+
+    freqs = dispersion_data[:,6:end] 
+    ordering_perms = [sortperm(freqs_at_fixed_k) for freqs_at_fixed_k in eachrow(freqs)]
+    # read mpb data, mostly as Strings (first column is Int)
+    untyped_data = readdlm(parentdir*"/output/plane_group_"*"$(sgnum)/"*calcname*"-symeigs.out", ',', quotes=true)
+    Nrows = size(untyped_data, 1)
+    # convert to typed format, assuming "Int, String, (ComplexF64 ...)" structure
+    ops = [Vector{SymOperation{D}}() for _ in kidxs] # indexed first across KVecs, then across SymOperations
+    symeigs =[Vector{Vector{ComplexF64}}() for _ in kidxs] # (--||--), then across bands
+    kidx = rowidx = 1
+    while kidx ≤ Nk
+        while rowidx ≤ Nrows && untyped_data[rowidx, 1] == kidx
+            op = SymOperation{D}(strip(untyped_data[rowidx, 2], '"'))
+            push!(ops[kidx], op)
+            # symmetry eigenvalues (frequency-sorted) at current `kidx` and current `op`
+            _symeigs = parse.(ComplexF64, @view untyped_data[rowidx, 3:end][ordering_perms[kidx]])
+            push!(symeigs[kidx], _symeigs)
+            rowidx += 1
+        end
+        kidx += 1
+    end
+    # build little groups and reconstruct labels
+    lgs⁰  = [primitivize(lg) for lg in values(get_littlegroups(sgnum, Val(D)))] # primitivizes & converts dict to vector...
+    idxs⁰ = if !flip_ksign
+        [findfirst(lg->isapprox(kvec(lg)(αβγ), constant(kv), atol=1e-6), lgs⁰) for kv in kvecs]
+    else # for trickery, to change the interpreted sign of k
+        [findfirst(lg->isapprox(kvec(lg)(αβγ), -constant(kv), atol=1e-6), lgs⁰) for kv in kvecs]
+    end
+    klabs = klabel.(getindex.(Ref(lgs⁰), idxs⁰))
+    lgs = [LittleGroup{D}(sgnum, kvecs[kidx], klabs[kidx], ops[kidx]) for kidx in kidxs]
+    return lgs, symeigs
+end
+
+"""
+$(TYPEDSIGNATURES)
+Load a symmetry data associated with `calcname` and compute the associated irrep for bands
+summed over `bandidxs`. If not supplied explicitly via `sgnum` and `D`, the space group number and dimension
+will be attempted to be inferred from the `calcname`.
+
+# Keyword arguments
+- `timereversal`: whether to use real (`true`, default) or complex irreps (`false`).
+- `isprimitive`: If the lattice in `calcname` is *not* in a primitive setting, this kwarg
+  must be set to `false` (this information is needed as part of a check that ensures operator 
+  equivalence across the little groups in `calcname` and those on from `get_lgirreps`).
+- `atol`: absolute tolerance passed to `find_representation` ($(Crystalline.DEFAULT_ATOL),
+  default)
+- `αβγ`: value used for pinning the little groups of lines, planes, and volumes to a
+  concrete k-vector. *Must* be identical to that used in setting up the calculation (e.g.,
+  as in Crystalline's `write_lgs_to_mpb`, where it defaults to `Crystalline.TEST_αβγ` as it
```

Comment:
Should probably not be `Crystalline's` above (since it's now in this package):

```suggestion
  as in `write_lgs_to_mpb`, where it defaults to `Crystalline.TEST_αβγ` as it
```